pintracker/optracker/operationtracker.go

```diff
@@ -209,18 +204,17 @@ func (opt *OperationTracker) GetAll(ctx context.Context) []*api.PinInfo {
 // CleanError removes the associated Operation, if it is
 // in PhaseError.
 func (opt *OperationTracker) CleanError(ctx context.Context, c cid.Cid) {
-	opt.mu.RLock()
-	defer opt.mu.RUnlock()
-	errop, ok := opt.operations[c.String()]
-	if !ok {
-		return
-	}
+	cidStr := c.String()
+
+	opt.mu.Lock()
+	defer opt.mu.Unlock()
 
-	if errop.Phase() != PhaseError {
+	errOp, ok := opt.operations[cidStr]
+	if !ok || errOp.Phase() != PhaseError {
 		return
 	}
 
-	opt.Clean(ctx, errop)
+	delete(opt.operations, cidStr)
```

Comment:
As far as I can see, this is changing behaviour and introducing a bug.

Reply:
There seems to be a double-locking issue, but it should only remove the operations when the pointer is the same.

Reply:
I changed this because of the double locking issue. How is this introducing a bug?

Reply:
You're right, sorry, I misread. In the pintracker PR I fully removed this method as it is not called from anywhere.